File: Baekjoon_Online_Judge/NO_02930/2930.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# -------------------------------------- #
# | Created by happiness96             | #
# | Year 2020                          | #
# | Month 11                           | #
# | Day 24                             | #
# | 2930 가위 바위 보                   | #
# -------------------------------------- #


if __name__ == "__main__":
    R = int(r_input())
    sg = r_input().rstrip()

    N = int(r_input())
    score = 0

    save = [[0, 0, 0] for _ in range(R)]

    for _ in range(N):
        fr = r_input().rstrip()

        for ind, val in enumerate(fr):
            save[ind]['RSP'.index(val)] += 1
        
        for ind, val in enumerate(sg):
            if val == fr[ind]:
                score += 1
            
            elif (val, fr[ind]) in [('S', 'P'), ('P', 'R'), ('R', 'S')]:
                score += 2

    max_score = 0

    for val in save:
        max_score += max(val[0] + val[1] * 2, val[1] + val[2] * 2, val[2] + val[0] * 2)
    
    print(score)
    print(max_score)

File: Baekjoon_Online_Judge/NO_02930/2930.cpp
#include <iostream>
#include <algorithm>
using namespace std;
/*
 -------------------------------------- 
 | Created by happiness96             | 
 | Year 2020                          | 
 | Month 11                           | 
 | Day 24                             | 
 | 2930 가위 바위 보                   | 
 -------------------------------------- 
*/

int main(){
    int R, N, score = 0, max_score = 0;
    string SG;

    cin >> R >> SG >> N;

    int save[R][3] = {{0,}};

    for(int i = 0; i < N; i ++){
        string fr;
        cin >> fr;

        for(int j = 0; j < R; j ++){
            int ind = 0;
            if(fr[j] == 'S') ind = 1;
            if(fr[j] == 'P') ind = 2;

            save[j][ind] ++;

            if(fr[j] == SG[j]) score ++;
            else if(SG[j] == 'R' && fr[j] == 'S' || SG[j] == 'S' && fr[j] == 'P' || SG[j] == 'P' && fr[j] == 'R') score += 2;
        }
    }

    for(int j = 0; j < R; j ++){
        max_score += max(max(save[j][0] + save[j][1] * 2, save[j][1] + save[j][2] * 2), save[j][2] + save[j][0] * 2);
    }

    cout << score << '\n' << max_score;

    return 0;
}